posts/ml-sagemaker.md

```diff
@@ -0,0 +1,220 @@
+
+# Machine Learning Process
+Typically Machine Learning process is an iterative process; it starts with identifying a use case to solve. Some of the steps involved in this process are as follows. This could change based on the use case you are trying to solve.
+
+* Identify and Obtain Data.
+* Pre-process and Prepare Data.
+* Train a Model using the Prepared Data.
+* Evaluate the Model.
+* Deploy the Model.
+* Obtain Feedback of the Model.
+
+Each of the above steps are iterative by themselves. Multiple iterations can happen during each of these steps depending on the data and performance of the model etc.
+
+![Machine Learning Process](https://raw.githubusercontent.com/msambaraju/blog-usa/master/images/2019/03/Machine_Learning.png)
+
+# AWS Sagemaker
+AWS Sagemaker is a fully managed AWS Machine Learning service which helps in building, training and deploying Machine Learning models. It has a rich set API's, built-in algorithms, integration with various popular libraries such as Tensorflow, PyTorch, SparkML etc. It also developers tools for authoring models, and hosted production environment for deploying the models.
+
+# Regression Model Implementation
+In this example we will develop a regression model to predict body fat percentage based on already labeled data with 14 parameters like age, height, weight, abdomen circumference etc. This data is widely available ([Dataset Location1](http://lib.stat.cmu.edu/datasets/bodyfat) or [Dataset Location2](http://wiki.stat.ucla.edu/socr/index.php/SOCR_Data_BMI_Regression#References)) on the internet. A more advanced process is to find only the relevant features or parameters to train the model.
+
+We will use the Jupyter Notebook authoring environment provided by Sagemaker to Prepare Data, Train and Evaluate a model, to Deploy and Test the model. The notebook environment also supports version control systems like CodeCommit or GitHub. You can upload any test data used by the Notebooks into the environment. The notebook lets you install any python libraries needed for your model using tools such as pip, run any git commands to push your changes to version control system.
+
+In order to use the Jupyter Notebook, we need to create a Notebook Instance by providing an instance type such as `ml.t2.medium`. Provide an IAM role with the proper access control during instance creation.
+
+Now create a new Jupyter Notebook to interactively Author, Train, Test and Deploy the Model.
```

Comment:
Author, Train, Test and Deploy the Model -> author, train, test and deploy the model